app.rb

```diff
@@ -0,0 +1,57 @@
+require './person'
+require './book'
+require './rental'
+require './teacher'
+require './student'
+
+# app file to hold all operations
+class App
+  attr_accessor :people, :books, :rentals
+
+  def initialize
+    @people = []
+    @books = []
+    @rentals = []
+  end
+
+  def list_all_books
+    @books.each do |book|
+      puts "#{book.id}: #{book.title} by #{book.author}"
+    end
+  end
+
+  def list_all_people
+    @people.each do |person|
+      puts "#{person.id}: #{person.name}"
+    end
+  end
+
+  def create_person(type, id, name, parent_permission)
+    if type == 'teacher'
+      @people << Teacher.new(id, name)
+    elsif type == 'student'
+      @people << Student.new(id, name, parent_permission: parent_permission)
+    else
+      puts 'Invalid person type.'
+    end
+  end
+
+  def create_book(id, title, author)
+    @books << Book.new(id, title, author)
+  end
+
+  def create_rental(id, person_id, book_id, rental_date)
+    @rentals << Rental.new(id, person_id, book_id, rental_date)
+  end
```

Comment:
- [ ] First, how can I create a rental when I have neither a person nor a book? This should not be the case. I was able to create a rental without a book or a person. And I suggest that in all your flow, you should include a success message with proper spacing so that the user's experience is improved. 

<img width="425" alt="image" src="https://github.com/Daniel-Kigozi/school_library/assets/84156714/75adbdff-c521-42db-af5d-37e78144cc32">


See an example of creating a rental. 

```ruby
def create_rental
    puts "Select a book from the following list by number"
    @books.each_with_index do |book, index|
      puts "#{index}) Title: \"#{book.title}\", Author: #{book.author}"
    end
    # *If needed, ask for parameters for the option.
    book_index = gets.chomp.to_i

    puts "\nSelect a person from the following list by number (not id)"
    @people.each_with_index do |person, index|
      puts "#{index}) [#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
    end
    # *If needed, ask for parameters for the option.
    person_index = gets.chomp.to_i

    print "\nDate: "
    # *If needed, ask for parameters for the option.
    date = gets.chomp

    @rentals << Rental.new(date, @books[book_index], @people[person_index])
    puts "Rental created successfully"
  end
```